src/librustc_middle/traits/chalk.rs

```diff
@@ -86,8 +73,8 @@ impl<'tcx> chalk_ir::interner::Interner for RustInterner<'tcx> {
     type InternedQuantifiedWhereClauses = Vec<chalk_ir::QuantifiedWhereClause<Self>>;
     type InternedVariableKinds = Vec<chalk_ir::VariableKind<Self>>;
     type InternedCanonicalVarKinds = Vec<chalk_ir::CanonicalVarKind<Self>>;
-    type DefId = RustDefId;
-    type InternedAdtId = RustDefId;
+    type DefId = DefId;
+    type InternedAdtId = DefId;
```

Comment:
Can we make this [`AdtDef<'tcx>`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.AdtDef.html) instead?

Reply:
Can't do this now since `InternedAdtDef` has a `Copy` bound.